plugins/registry.js

```diff
@@ -584,4 +584,58 @@ export default class PluginRegistry {
 
         return {id, showRHSPlugin: showRHSPlugin(id), hideRHSPlugin: hideRHSPlugin(id), toggleRHSPlugin: toggleRHSPlugin(id)};
     }
+
+    // Register a Needs Team component by providing a route past /:team/ to be displayed at.
+    // Accepts the following:
+    // - route - The route to be displayed at.
+    // - component - A react component to display.
+    // Returns:
+    // - id: a unique identifier
+    registerNeedsTeamComponent(route, component) {
+        const id = generateId();
+        let fixedroute = route.trim();
+        if (fixedroute[0] === '/') {
+            fixedroute = fixedroute.substring(1);
+        }
+
+        store.dispatch({
+            type: ActionTypes.RECEIVED_PLUGIN_COMPONENT,
+            name: 'NeedsTeamComponent',
+            data: {
+                id,
+                pluginId: this.id,
+                component,
+                route: fixedroute,
+            },
+        });
+
+        return id;
+    }
+
+    // Register a component to be displayed at a custom route under /plug/
+    // Accepts the following:
+    // - route - The route to be displayed at.
+    // - component - A react component to display.
+    // Returns:
+    // - id: a unique identifier
+    registerCustomRoute(route, component) {
+        const id = generateId();
+        let fixedroute = route.trim();
```

Comment:
Minor nit, but the casing on this should be `fixedRoute`

Reply:
I don't know why I keep doing this. I keep not camelCasing stuff when I am typing.